lib/zinnia.js

```diff
@@ -171,14 +185,16 @@ export async function run ({
             type: 'info',
             message: 'Updated Zinnia module source files, restarting...'
           })
-          childProcess.kill()
+          for (const childProcess of childProcesses) {
+            childProcess.kill()
+          }
           return
         }
       }
     })(),
     (async () => {
       try {
-        await childProcess
+        await Promise.race(childProcesses)
```

Comment:
IIUC, this will return as soon as the first process exits, and the other promises will stay running in the background. What is going to happen with those promises? In particular, if the promise slower to finish ends up being rejected, how will we handle that error?

Looking from a higher-level perspective, we need to decide how to handle the following cases:

- Both (all) modules start and happily run
- One module starts, but the other crashes
- Both (all) modules crashes

The answers are not clear to me from reading the code.

Reply:
See the current version of the code (I added some changes after your feedback). When one process exits, all others will be killed too.

In the previous version of the code, once the first child process closed stdout, all others were killed.

This means that if one Zinnia process is unhealthy, all will be taken down. I think we can iterate this and decouple the processes completely, if we see that this is causing problems.

Reply:
> When one process exits, all others will be killed too.
> This means that if one Zinnia process is unhealthy, all will be taken down. I think we can iterate this and decouple the processes completely, if we see that this is causing problems.

SGTM 👍🏻 

Let's ship a simple version and then iterate based on observing real-world behaviour.